util.go

```diff
@@ -101,6 +101,73 @@ func BuildGarpPkt(ip net.IP, mac net.HardwareAddr, vlanID uint16) *openflow13.Pa
 	return pktOut
 }
 
+func ipChecksum(data []byte) uint16 {
+	csum := uint32(0)
+	length := len(data) - 1
+	for i := 0; i < length; i += 2 {
+		csum += uint32(data[i]) << 8
+		csum += uint32(data[i+1])
+	}
+	if len(data)%2 == 1 {
+		csum += uint32(data[length]) << 8
+	}
+	for csum > 0xffff {
+		csum = (csum >> 16) + (csum & 0xffff)
+	}
+	return ^uint16(csum)
+}
+
+func buildDnsForwardPkt(ethInPkt *protocol.Ethernet) *protocol.Ethernet {
+	ipPkt := ethInPkt.Data.(*protocol.IPv4)
+	ethPkt := protocol.NewEthernet()
+	ethPkt.HWSrc = ethInPkt.HWSrc
+	ethPkt.HWDst = ethInPkt.HWDst
+	ethPkt.VLANID.VID = 4093 // magic vlan
```

Comment:
This vlan seems to be used throughout the code for flow installation. Would be better to have a const var for this.